lua/ttt2/libraries/playermodels.lua

```diff
@@ -0,0 +1,145 @@
+---
+-- A module that contains all functions related to playermodels
+-- @author Mineotopia
+
+if SERVER then
+	AddCSLuaFile()
+
+	util.AddNetworkString("TTT2UpdateSelectedModels")
+end
+
+local pairs = pairs
+local playerManagerAllValidModels = player_manager.AllValidModels
+local utilPrecacheModel = util.PrecacheModel
+local stringSplit = string.Split
+local tableRemove = table.remove
+local tableConcat = table.concat
+local tableHasValue = table.HasValue
+local RunConsoleCommand = RunConsoleCommand
+local mathRandom = math.random
+
+playermodels = playermodels or {}
+
+---
+-- Adds a new model name to the list of the selected playermodes.
+-- @param string name The name of the model
+-- @realm shared
+function playermodels.AddSelectedModel(name)
+	net.Start("TTT2UpdateSelectedModels")
+	net.WriteBool(true)
+	net.WriteString(name)
+	net.SendToServer()
+end
+
+---
+-- Removes a model by its name from the list of the selected playermodes.
+-- @param string name The name of the model
+-- @realm shared
+function playermodels.RemoveSelectedModel(name)
+	net.Start("TTT2UpdateSelectedModels")
+	net.WriteBool(false)
+	net.WriteString(name)
+	net.SendToServer()
+end
+
+if CLIENT then
+	local callbackCache = {}
+
+	local function Inernal_OnDataAvailable(_, modelString)
+		for i = 1, #callbackCache do
+			local Callback = callbackCache[i]
+
+			if not isfunction(Callback) then continue end
+
+			Callback(stringSplit(modelString, ","))
+		end
+
+		callbackCache = {}
+	end
+
+	---
+	-- Rerturns an indexed table with all the models that are in the selction pool. Sends
+	-- a request to the server and the data is provided in the callback function.
+	-- @param function OnDataAvailable The callback function that is called once the data is available
+	-- @realm client
+	function playermodels.GetSelectedModels(OnDataAvailable)
+		callbackCache[#callbackCache + 1] = OnDataAvailable
+
+		cvars.ServerConVarGetValue("ttt2_selected_playermodels", Inernal_OnDataAvailable)
+	end
+end
+
+if SERVER then
+	---
+	-- @realm server
+	local cvSelectPlayermodels = CreateConVar("ttt2_selected_playermodels", "css_phoenix,css_arctic,css_guerilla,css_leet", {FCVAR_ARCHIVE})
```

Comment:
Please use the database or any other medium, i don't know if convars are unlimited, but they are for sure not optimized for such usage and it is hard to manage.